### PR TITLE
Ticket2813 riken changeover logic

### PR DESCRIPTION
**Pull request by Tom-Willemsen**
### Description of work

Adds a coordinator IOC.

This IOC is intended to be reusable in different configurations but initially is being used for two (very similar) tasks:
- Riken port changeover
- Riken RB2 mode change

### To test

https://github.com/ISISComputingGroup/IBEX/issues/2813

### Acceptance criteria

- Changeovers are started when input PV is changed (currently pointing at SIMPLE)
- Waits for relevant power supplies to be OFF
- Sets DISP field on power setpoints
- Allows changeover (currently pointing at SIMPLE)
- Waits for changeover to be complete
- Re-enables DISP on power PVs

- Process can be safely interrupted at any point in the above process.

---

#### Code Review

- [ ] **Pertitent information and a copy of the manual has been stored in the [wiki](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Specific-Device-IOC)**
- [ ] Does the IOC conform to IBEX standards?
    - [PV naming](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/PV-Naming).
    - [Disable records](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Disable-records)
    - [Record simulation](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Record-Simulation)
    - [Finishing touches](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/IOC-Finishing-Touches)
- [x] If an OPI has been modified, does it conform to the [style guidelines](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/OPI-Creation)?
- [x] Have the changes been recorded appropriately in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev)?

### Functional Tests

- IOC responds correctly in:
    - [ ] Devsim mode
    - [ ] Recsim mode
    - [ ] Real device, if available
- [x] Supplementary IOCs (`..._0n` where `n>1`) run correctly
- [ ] Log files do not report undefined macros (serach for `macLib: macro` to find instances of `macLib: macro [macro name] is undefined...`

### Final steps

- [ ] Update the IOC submodule in the main EPICS repo. See [Git workflow](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Git-workflow) page for details.
- [ ] Reviewer has moved the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev) entry for this ticket in the "Changes merged into master" section

### CODE REVIEW

File: COORD/configure/RELEASE [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# RELEASE - Location of external support modules
```
Comment (DominicOram) [inline COORD/configure/RELEASE]: Can you avoid the use of MASTER_RELEASE as per https://github.com/ISISComputingGroup/IBEX/issues/2320. See https://github.com/ISISComputingGroup/EPICS-ioc/blob/master/ISISDAE/configure/RELEASE for an example. Add this IOC to the ticket when done.

File: COORD/iocBoot/iocCOORD-IOC-01/st_riken_port_changeover.cmd [hunk 2]
```diff
@@ -0,0 +1,8 @@
+epicsEnvSet(PC_OK_TO_RUN_PSUS,$(MYPVPREFIX)SIMPLE:VALUE1)
+epicsEnvSet(PC_ALLOW_CHANGEOVER,$(MYPVPREFIX)SIMPLE:VALUE2)
+epicsEnvSet(PC_PSU_DISABLE,$(MYPVPREFIX)$(IOCNAME):PC:PSUS:DISABLE)
+epicsEnvSet(PC_PSU_POWER,$(MYPVPREFIX)$(IOCNAME):PC:PSUS:POWER)
+
+dbLoadRecords("$(TOP)/db/riken_changeover.db","PV_PREFIX=$(MYPVPREFIX),P=$(MYPVPREFIX)$(IOCNAME):,OK_TO_RUN_PSUS=$(PC_OK_TO_RUN_PSUS),PSU_DISABLE=$(PC_PSU_DISABLE),PSU_POWER=$(PC_PSU_POWER)")
+dbLoadRecords("$(TOP)/db/riken_port_changeover_psus.db","PV_PREFIX=$(MYPVPREFIX),P=$(MYPVPREFIX)$(IOCNAME):,OK_TO_RUN_PSUS=$(PC_OK_TO_RUN_PSUS),PSU_DISABLE=$(PC_PSU_DISABLE),PSU_POWER=$(PC_PSU_POWER)")
+dbLoadRecords("$(TOP)/db/riken_changeover_groups.db","PV_PREFIX=$(MYPVPREFIX),P=$(MYPVPREFIX)$(IOCNAME):,OK_TO_RUN_PSUS=$(PC_OK_TO_RUN_PSUS),PSU_DISABLE=$(PC_PSU_DISABLE),PSU_POWER=$(PC_PSU_POWER)")
```
Comment (DominicOram) [inline COORD/iocBoot/iocCOORD-IOC-01/st_riken_port_changeover.cmd]: For readability, only pass the macros which are actually required.

Comment (DominicOram) [reply COORD/iocBoot/iocCOORD-IOC-01/st_riken_port_changeover.cmd]: In which case do we need to set them as EPICS env variables at all or can we just pass them straight to the seq in the st.cmd?

**Comment by DominicOram** [comment]
Seems to work as described, some small issues:
* Please write a page on the wiki about this IOC, both the things you think could be used in other situations and the bits specific to this case.
* This isn't a requirement of this ticket but would it be useful to add an OPI for this functionality? e.g. turn off all three power supplies with one button, give feedback on what the next steps are. @KathrynBaker?
* As discussed:
     * Can you pull the PLC into a macro?
     * Only one group is needed

File: COORD/configure/RELEASE (hunk 1 above)
Comment (Tom-Willemsen) [reply COORD/configure/RELEASE]: done.

File: COORD/iocBoot/iocCOORD-IOC-01/st_riken_port_changeover.cmd (hunk 2 above)
Comment (Tom-Willemsen) [reply COORD/iocBoot/iocCOORD-IOC-01/st_riken_port_changeover.cmd]: The reason you can't pass them straight to seq in the individual st.cmds is that starting a sequence program before `iocInit` is called is a bad idea (undefined behaviour?)

**Comment by Tom-Willemsen** [comment]
Review comments addressed. Documentation added.

**Pull request merged**